x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java

```diff
@@ -705,9 +707,23 @@ private CloseableHttpAsyncClient createHttpClient() {
                     .setConnectionRequestTimeout(Math.toIntExact(realmConfig.getSetting(HTTP_CONNECTION_READ_TIMEOUT).getSeconds()))
                     .setSocketTimeout(Math.toIntExact(realmConfig.getSetting(HTTP_SOCKET_TIMEOUT).getMillis()))
                     .build();
+                final long userConfiguredKeepAlive = realmConfig.getSetting(HTTP_CONNECTION_POOL_TTL).millis();
                 HttpAsyncClientBuilder httpAsyncClientBuilder = HttpAsyncClients.custom()
                     .setConnectionManager(connectionManager)
-                    .setDefaultRequestConfig(requestConfig);
+                    .setDefaultRequestConfig(requestConfig)
+                    .setKeepAliveStrategy((response, context) -> {
+                        var serverKeepAlive = DefaultConnectionKeepAliveStrategy.INSTANCE.getKeepAliveDuration(response, context);
+                        final long actualKeepAlive;
+                        if (serverKeepAlive == -1) {
+                            actualKeepAlive = userConfiguredKeepAlive;
+                        } else if (userConfiguredKeepAlive == -1) {
+                            actualKeepAlive = serverKeepAlive;
+                        } else {
+                            actualKeepAlive = Math.min(serverKeepAlive, userConfiguredKeepAlive);
+                        }
+                        LOGGER.debug("effective HTTP connection keep-alive: [{}]ms", actualKeepAlive);
+                        return actualKeepAlive;
+                    });
```

Comment:
I think we need to come up with a test strategy for this. 
We could do it with a mock HTTP Server and ensure that the connection is pulled down, but that seems like a lot of work.
Is it possible to check the value configured on the client itself?

Reply:
I came up a way for testing the changes for both the configured TTL and the actual behaviour of "not reusing the same connection". Please see https://github.com/elastic/elasticsearch/pull/87773/commits/058872f1fd3f78f383f1517ac4917ed1a007330f